code/tick/feed.q

```diff
@@ -31,15 +31,37 @@ randomize:{value "\\S ",string "i"$0.8*.z.p%1000000000}
 rnd:{0.01*floor 0.5+x*100}
 vol:{10+`int$x?90}
 
-/ randomize[]
-\S 235721
+randomize[]
+/ \S 235721
 
-/ =========================================================
+/ =========================================================================================
+/ generate weights to stop even distribution of counts and sizes
+
+weight:0.3 0.6 1.8 1.1 1.2 1.6 0.9 0.4 0.7 1.4
+
+/ assign multipliers to skew size columns
+volmap:s!-10?weight
+bidmap:s!-10?weight
+askmap:s!-10?weight
+
+/ returns list where count of each item is given by random permutation of integer weights
+skewitems:{[weights;items] raze (#) .' weights,' neg[count items]?items}
+
+/ skew sym counts with weighted list of indices
+weightedsyms:skewitems[`long$weight*10;til cnt] / 100 weighted indices when cnt=10
+
+/ assign skewed side and src lists to determine probabilities of appearing
+sideweight:cnt?{x,cnt-x}'[1+til 4]
+sidemap:s!skewitems[;side] each sideweight
+srcweight:1 2 3 4
```

Comment:
are these deliberately not randomised?

Reply:
skewitems randomises the items each time it's ran so the weights are randomly assigned

Reply:
OK, does it still work if there are more than 4 sources? Or should this be based on `count[src]`?

Reply:
ok yeah will change to 
```1+til count src```